cmd/edit.go

```diff
@@ -0,0 +1,155 @@
+			continue
+		}
+
+		parentFlags := kind.GetParentFlag()
+		parentQueryFlags := kind.GetParentQueryFlag()
+		parentFlagValue := make([]*string, len(parentFlags))
+		parentQueryFlagValue := make([]*string, len(parentQueryFlags))
+
+		kindCmd := &cobra.Command{
+			Use:     use,
+			Short:   "Edit resource of kind " + name,
+			Args:    args,
+			Long:    `Edit the specified resource in a text editor and apply changes after saving`,
+			Aliases: buildAlias(name),
+			Run: func(cmd *cobra.Command, args []string) {
+				parentValue := make([]string, len(parentFlagValue))
+				parentQueryValue := make([]string, len(parentQueryFlagValue))
+				for i, v := range parentFlagValue {
+					parentValue[i] = *v
+				}
+				for i, v := range parentQueryFlagValue {
+					parentQueryValue[i] = *v
+				}
+
+				resourceName := args[0]
+
+				// Get the resource
+				var result resource.Resource
+				var err error
+
+				if isGatewayKind {
+					result, err = gatewayApiClient().Describe(&kind, parentValue, parentQueryValue, resourceName)
+				} else {
+					result, err = consoleApiClient().Describe(&kind, parentValue, parentQueryValue, resourceName)
+				}
+
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error describing resource: %s\n", err)
+					os.Exit(1)
+				}
+
+				// Create a temporary file with the resource YAML
+				tmpFile, err := os.CreateTemp("", fmt.Sprintf("conduktor-edit-%s-%s-*.yaml", name, resourceName))
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error creating temporary file: %s\n", err)
+					os.Exit(1)
+				}
+				defer os.Remove(tmpFile.Name()) // Clean up temp file
+
+				// Write the resource to the temp file
+				w := bufio.NewWriter(tmpFile)
+				_, err = w.WriteString(AutoApplyWarningMessage)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error writing to temporary file: %s\n", err)
+					os.Exit(1)
+				}
+				_, err = w.WriteString("---\n")
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error writing to temporary file: %s\n", err)
+					os.Exit(1)
+				}
+
+				var data orderedjson.OrderedData
+				err = json.Unmarshal(result.Json, &data)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error unmarshalling resource JSON: %s\n", err)
+					os.Exit(1)
+				}
+
+				err = printutils.PrintResourceLikeYamlFile(w, data)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error writing resource YAML: %s\n", err)
+					os.Exit(1)
+				}
+
+				err = w.Flush()
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error flushing temporary file: %s\n", err)
+					os.Exit(1)
+				}
+				tmpFile.Close()
+
+				// Open the file in an editor
+				err = runEditor(tmpFile.Name())
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Editor error: %s\n", err)
+					os.Exit(1)
+				}
+
+				content, err := os.ReadFile(tmpFile.Name())
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Error reading modified file: %s\n", err)
+					os.Exit(1)
+				}
+
+				// If file is empty, don't apply
+				if len(content) == 0 {
```

Comment:
we could also skip apply if no change was made to the resource by reading file content and comparing it with  OrderedData